libs/dsb-client-gateway-secrets-engine/src/lib/service/azure-key-vault.service.ts

```diff
@@ -0,0 +1,236 @@
+  implements OnModuleInit
+{
+  private readonly logger = new Logger(AzureKeyVaultService.name);
+
+  protected client: SecretClient;
+  protected readonly prefix: string;
+
+  constructor(protected readonly configService: ConfigService) {
+    super();
+    this.prefix = this.configService.get('AZURE_KEY_PREFIX', 'ddhub/');
+  }
+
+  @Span('azure_kv_onModuleInit')
+  public async onModuleInit(): Promise<void> {
+    const vaultName = this.configService.get('AZURE_VAULT_NAME');
+    const url = `https://${vaultName}.vault.azure.net`;
+    const credential = new DefaultAzureCredential();
+
+    this.client = new SecretClient(url, credential);
+
+    this.logger.log('Azure Key Vault Service initialized');
+  }
+
+  @Span('azure_kv_setRSAKey')
+  public async setRSAPrivateKey(
+    privateKey: string
+  ): Promise<KeyVaultSecret | null> {
+    const name = this.encodeAzureKey(`${this.prefix}${PATHS.RSA_KEY}`);
+    return this.client
+      .setSecret(name, privateKey)
+      .then((response) => {
+        this.logger.log(`Successfully set RSA private key: ${name}`);
+        return response;
+      })
+      .catch(async (err) => {
+        this.logger.error(err.message);
+        return null;
+      });
+  }
+
+  @Span('azure_kv_getRSAKey')
+  public async getRSAPrivateKey(): Promise<string | null> {
+    return this.client
+      .getSecret(this.encodeAzureKey(`${this.prefix}${PATHS.RSA_KEY}`))
+      .then(({ value }) => {
+        if (!value || value === '' || value === '""') {
+          throw new Error('Secret does not contain valid value');
+        }
+        return value;
+      })
+      .catch((err) => {
+        this.logger.error(err.message);
+        return null;
+      });
+  }
+
+  @Span('azure_kv_setCertificateDetails')
+  public async setCertificateDetails({
+    caCertificate,
+    certificate,
+    privateKey,
+  }: CertificateDetails): Promise<KeyVaultSecret[]> {
+    const commands = [
+      this.client.setSecret(
+        this.encodeAzureKey(`${this.prefix}${PATHS.CERTIFICATE_KEY}`),
+        privateKey
+      ),
+      this.client.setSecret(
+        this.encodeAzureKey(`${this.prefix}${PATHS.CERTIFICATE}`),
+        certificate
+      ),
+    ];
+
+    if (caCertificate) {
+      commands.push(
+        this.client.setSecret(
+          this.encodeAzureKey(`${this.prefix}${PATHS.CA_CERTIFICATE}`),
+          caCertificate
+        )
+      );
+    }
+
+    const responses = await Promise.allSettled(
+      commands.map((command) =>
+        command
+          .then((response) => response)
+          .catch((err) => {
+            throw new Error(
+              JSON.stringify({
+                error: err,
+              })
+            );
+          })
+      )
+    );
+
+    const errors = responses.filter(
+      ({ status }) => status === 'rejected'
+    ) as PromiseRejectedResult[];
+
+    if (errors.length > 0) {
+      this.logger.error(errors.map(({ reason }) => reason.message).join(', '));
```

Comment:
shouldn't you throw erroror something like that?

Reply:
If I throw an error what would be the expected state of the other `setSecret`? Updated or rollback?

Reply:
@hejkerooo What are your thoughts with error handling here?

Reply:
@nik-ewf Rollback